client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,134 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	source: 'email-comparison' | 'email-management';
+	usePromoCard: boolean;
+};
+
+const buildQueryString = ( parameters = {} ) =>
+	parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = emailManagementPurchaseNewEmailAccount(
+			selectedSite?.slug ?? '',
+			domain?.name ?? '',
+			null,
+			'login-redirect'
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const contactOwnerUrl = `https://privatewho.is/${ buildQueryString( {
+		s: domain?.name,
+	} ) }`;
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'contact' | 'login' | 'support' ) => {
+		const properties = {
+			action: eventType,
+		};
+
+		recordTracksEvent( `calypso_email_providers_nonowner_click`, properties );
+	};
+
+	const translateOptions = {
+		components: {
+			loginLink: <a href={ loginUrl } onClick={ () => onClickLink( 'login' ) } rel="external" />,
+			reachOutLink: isPrivacyAvailable ? (
+				<a
+					href={ contactOwnerUrl }
+					onClick={ () => onClickLink( 'contact' ) }
+					rel="noopener noreferrer"
+					target="_blank"
+				/>
+			) : (
+				<></>
+			),
+			contactSupportLink: (
+				<a
+					href="https://wordpress.com/help/contact"
+					onClick={ () => onClickLink( 'support' ) }
+					rel="noopener noreferrer"
+					target="_blank"
+				/>
+			),
+		},
+		args: {
+			ownerUserName,
+			selectedDomainName: domain?.name,
+		},
+	};
+
+	let reasonText: TranslateResult | null = null;
+
+	if ( source === 'email-comparison' ) {
+		if ( ownerUserName ) {
+			reasonText = translate(
+				'Email service can only be purchased by %(ownerUserName)s, ' +
+					'who is the owner of %(selectedDomainName)s. ' +
```

Comment:
Can we bold the username as well as the domain name so they stand out more? The username within the hyperlink can be left as is.